File: content/examples/other/leetcode/problems/rotate-image/example-2/solution.js
function rotate(matrix) {
  const n = matrix.length;
  // transpose
  for (let r = 0; r < n; r += 1) {
    for (let c = r + 1; c < n; c += 1) {
      [matrix[r][c], matrix[c][r]] = [matrix[c][r], matrix[r][c]];
    }
  }
  // reverse rows
  for (let r = 0; r < n; r += 1) matrix[r].reverse();
  return matrix;
}

if (require.main === module) {
  console.log(rotate([[1, 2, 3], [4, 5, 6], [7, 8, 9]]));
}

module.exports = { rotate };

File: content/examples/other/leetcode/problems/rotate-image/example-2/solution.py
def rotate(matrix):
    n = len(matrix)
    for r in range(n):
        for c in range(r + 1, n):
            matrix[r][c], matrix[c][r] = matrix[c][r], matrix[r][c]
    for r in range(n):
        matrix[r].reverse()
    return matrix


if __name__ == "__main__":
    print(rotate([[1, 2, 3], [4, 5, 6], [7, 8, 9]]))
